### ISSUE TITLE
support for localisation

### ISSUE THREAD

**Issue by devzwf** [ellite/Wallos#5]
support for different Language would be awesome :)

**Comment by ellite** [comment]
Never did anything like that. At the moment wouldn't even know where to start.
But I will add it to my list and have a look how I could do that.
Thank you for the suggestion.

**Comment by devzwf** [comment]
i see
when you ready i can do 1 or 2 language for you :)

**Comment by ellite** [comment]
Appreciate it. Thank you.

**Comment by CrowMash** [comment]
French native speaker here if you need help

**Comment by devzwf** [comment]
i am also french speaker as well
waiting on @ellite to implement so we can add some language :)

**Comment by ellite** [comment]
This is finally merged. Was a long time in progress.
https://github.com/ellite/Wallos/commit/d7366dcfb0413b99279d867164da438779454a59

For now released as docker images on the edge and dev tags. Will be creating the stable and latest images soon if I don't find any big issue with it.

**Comment by devzwf** [comment]
great will have a look

**Comment by devzwf** [comment]
i create a PR for the french #86 
@CrowMash  please you can correct if need...... 
thanks

**Comment by ellite** [comment]
That was fast. Thank you for the contribution.

**Comment by CrowMash** [comment]
Wow that was fast! Great work!

Found only two typos:

https://github.com/ellite/Wallos/blob/a4926ba9ba6ee8302b22aa108240d1b401c19bd4/includes/i18n/en.php#L32
Should be "Name"

https://github.com/ellite/Wallos/blob/a4926ba9ba6ee8302b22aa108240d1b401c19bd4/includes/i18n/fr.php#L13
Should be "échoué"

**Comment by ellite** [comment]
Thank you. I will fix those soon.

**Comment by CrowMash** [comment]
Typo here too:
https://github.com/ellite/Wallos/blob/a4926ba9ba6ee8302b22aa108240d1b401c19bd4/includes/list_subscriptions.php#L17

**Comment by CrowMash** [comment]
I think it's causing the app to crash:

```
Warning
: Undefined variable $i18n in
/var/www/html/includes/i18n/getlang.php
on line
17


Fatal error
: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, null given in /var/www/html/includes/i18n/getlang.php:17 Stack trace: #0 /var/www/html/includes/list_subscriptions.php(17): translate('YEarly', Array) #1 /var/www/html/index.php(79): getBillingCycle(4, 1, Array) #2 {main} thrown in
/var/www/html/includes/i18n/getlang.php
on line
17
```

**Comment by ellite** [comment]
Thank you. I'll fix it as well.

**Comment by devzwf** [comment]
Thanks @CrowMash  i am far from perfect specially "en ortographe" Lol
@ellite let me know if you need me to fix or you will

**Comment by ellite** [comment]
I can fix after work. If you're faster, you can go ahead 👌

**Comment by devzwf** [comment]
PR #88 filed

**Comment by ellite** [comment]
@CrowMash it shouldn't break, I've built a fallback for that, but it was buggy.
https://github.com/ellite/Wallos/commit/2f3fb1ee8ba369a4774502c4af14ae6a4b375669 should fix it.

**Comment by devzwf** [comment]
nice i will test this PM

**Comment by CrowMash** [comment]
Working great now!

I found small issues or possible improvements.

No translation here:
https://github.com/ellite/Wallos/blob/de5f03b8a8fa208eece6f6b239b916762ee2632d/settings.php#L208

In french, we put a space between text and ":" (explained here [polyglotclub](https://polyglotclub.com/wiki/Language/French/Grammar/Punctuation#Summary)), here is an example:
https://github.com/ellite/Wallos/blob/de5f03b8a8fa208eece6f6b239b916762ee2632d/settings.php#L182

Because of word length, there is a line break in the menu:
![image](https://github.com/ellite/Wallos/assets/9526631/9d106be3-4eb6-4171-8f95-64b71ae0e357)

**Comment by ellite** [comment]
Thanks. I'll have a look at the missing translation.
The issue with the menu is that the old version of the css file is cached, clearing the cache should fix it. 
<img width="164" alt="Screenshot 2024-01-26 at 11 40 39" src="https://github.com/ellite/Wallos/assets/3831180/fbd71895-af30-4e0f-8069-32fa358c8050">

**Comment by CrowMash** [comment]
Thanks it solved the issue!

**Comment by devzwf** [comment]
@ellite i will put a PR #92  for the missing translation , but how do you want to handle missing translation in the language file ?
i mean to notify maintainer that the en files change ?
let me know as i found a few place where translation need to be added

**Comment by ellite** [comment]
Maybe Chatgpt and tag the original translators after merge if they feel it needs fixing?

**Comment by devzwf** [comment]
I thing you should have a file with maintainers , and an action who will tag maintainers every time the following file change :

- includes/i18n/en.pgp 
- scripts/i18n/en.js


or something along the line .
tagging known maintainer for idea :

- FR : @CrowMash   @devzwf 
- DE : @joek-bs 
- CN: @QuietBlade @AnotiaWang 
- JP: @tkymmm 
- PT: @ellite 
- EN: @ellite 
Note: whole we are at it , you should start maintain a changelog and have released see ( #94  ) 

Just my 2c
